### PR TITLE
Improve helium-csharp Logging

### LINKED ISSUE #24: Improve helium-csharp Logging
- Tracks Helium issue 90

### PR DESCRIPTION
**Pull request by bartr**
Closes #24 

- update logging to make correlation easier
- log individual health check test results
- log aggregate health check test results
- add IP address to log

### CODE REVIEW

File: src/app/Controllers/HealthzController.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        [Produces("application/json")]
+        [ProducesResponseType(typeof(CosmosHealthCheck), 200)]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "Method logs and handles all exceptions")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Globalization", "CA1303:Do not pass literals as localized parameters", Justification = "log messages")]
+        public async System.Threading.Tasks.Task<IActionResult> RunHealthzJsonAsync()
+        {
+            _logger.LogInformation(nameof(RunHealthzAsync));
+
+            DateTime dt = DateTime.UtcNow;
+
```
Comment (aflinchb) [inline src/app/Controllers/HealthzController.cs:84]: DateTime object isn't used here

File: src/app/Middleware/requestLogger.cs [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
         public bool Log4xx { get; set; } = true;
         public bool Log5xx { get; set; } = true;
+        public double TargetMs { get; set; } = 250;
     }
 
+    /// <summary>
+    /// Middleware extension to make registering Logger easy
+    /// 
+    /// Note: Logger should be on of the first things registered in DI
+    /// </summary>
```
Comment (aflinchb) [inline src/app/Middleware/requestLogger.cs:28]: nit: typo -- "on" should be "one" :)

File: src/app/Controllers/HealthzController.cs (hunk 1 above)
Comment (bartr) [reply src/app/Controllers/HealthzController.cs:84]: removed

File: src/app/Middleware/requestLogger.cs (hunk 2 above)
Comment (bartr) [reply src/app/Middleware/requestLogger.cs:28]: fixed typo

**Review by aflinchb — APPROVED** [review]

**Pull request merged**